src/menu.rb

```diff
@@ -0,0 +1,175 @@
+require './book'
+require './rental'
+require './student'
+require './teacher'
+require './persons'
+
+class App
+  def initialize
+    @book_list = []
+    @people_list = []
+    @rental_list = []
+    @user_menu = "
+			Please choose an option by entering a number:
+			1 - List all books
+			2 - List all people
+			3 - Create a person
+			4 - Create a book
+			5 - Create a rental
+			6 - List all rentals for a given person id
+			7 - Exit
+		"
+  end
+
+  def menu
+    loop do
+      puts @user_menu
+      @input = gets.chomp.to_i
+      users_selection(@input)
+      break if @input == 7
+    end
+  end
+
+  def users_selection(users_input)
+    case users_input
+    when 1
+      list_books
+    when 2
+      all_people
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      list_all_rentals_by_id
+    else
+      puts 'Thank you for using this app'
+    end
+  end
+
+  def list_books
+    if @book_list.empty?
+      puts 'There are no book in the library yet!!'
+    else
+      @book_list.each do |book|
+        puts "Title: #{book.title}\n Author: #{book.author}\n"
+      end
+    end
+  end
+
+  def all_people
+    if @people_list.empty?
+      puts 'There are not people yet'
+    else
+      @people_list.each do |person|
+        puts "Name: #{person.name}\n Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_student
+    puts "Enter age:\n"
+    age = gets.chomp.to_i
+    puts 'Age must be a number' unless (age.is_a? Integer) && age > 1 && age < 100
+
+    puts 'Name:'
+    name = gets.chomp
+    return puts 'Name too long' unless name.length.positive? && name.length < 50
+
+    puts "Parent authorization? [Y/N]\n"
+    parent_permission = gets.chomp
+    parent_permission = parent_permission.downcase == 'y'
+
+    @people_list << Student.new(id, age, name, parent_permission: parent_permission)
+    puts "Student created successfully\n"
+  end
+
+  def create_teacher
+    puts "Age: \n"
+    age = gets.chomp.to_i
+    puts 'Age must be a number' unless (age.is_a? Integer) && age.positive? && age < 100
+
+    puts "Name: \n"
+    name = gets.chomp
+    puts 'Name too long' unless name.length.positive? && name.length < 100
+
+    puts "Specialization: \n"
+    specialiation = gets.chomp
+    specialiation = specialiation.downcase
+    @people_list << Teacher.new(id, age, name, specialiation)
+    puts 'Teacher created successfully'
+  end
+
+  def create_book
```

Comment:
- [ ] I tried creating a book and I am getting an error kindly fix that please.

![image](https://user-images.githubusercontent.com/94551484/216785973-d9a46c2d-eb22-4175-8078-d25ef08dc0a2.png)